proxy/core/src/main/scala/io/cloudstate/proxy/eventing/EventingManager.scala

```diff
@@ -0,0 +1,124 @@
+package io.cloudstate.proxy.eventing
+
+import akka.{Done, NotUsed}
+import akka.actor.{ActorSystem, Cancellable}
+import akka.http.scaladsl.model.{HttpEntity, HttpRequest, HttpResponse, StatusCodes}
+import akka.http.scaladsl.model.Uri.Path
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Flow, Keep, Merge, RunnableGraph, Sink, Source}
+import akka.parboiled2.util.Base64
+
+import io.cloudstate.protocol.entity.{ClientAction, EntityDiscoveryClient, Failure, Reply, UserFunctionError}
+import io.cloudstate.proxy.{Serve, UserFunctionRouter}
+import io.cloudstate.proxy.EntityDiscoveryManager.ServableEntity
+import io.cloudstate.proxy.protobuf.{Options, Types}
+import io.cloudstate.proxy.entity.{UserFunctionCommand, UserFunctionReply}
+
+import io.cloudstate.eventing.{Eventing, EventingProto}
+
+import com.google.protobuf.{ByteString => ProtobufByteString}
+import com.google.protobuf.any.{Any => ProtobufAny}
+import com.google.protobuf.Descriptors.{FileDescriptor, MethodDescriptor}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+import com.typesafe.config.Config
+
+import org.slf4j.{Logger, LoggerFactory}
+
+import Serve.{CommandHandler}
+
+trait EventingSupport {
+  def createSource(sourceName: String, handler: CommandHandler): Source[ProtobufByteString, Future[Cancellable]]
+  def createDestination(destinationName: String, handler: CommandHandler): Flow[ProtobufByteString, AnyRef, NotUsed]
+
+  final def serve(eventing: Eventing,
+                  commandHandler: CommandHandler,
+                  entityDiscoveryClient: EntityDiscoveryClient,
+                  log: Logger): Source[AnyRef, Future[Cancellable]] =
+    createSource(eventing.in, commandHandler)
+      .map(commandHandler.serializer.parse)
+      .via(commandHandler.flowUsing(entityDiscoveryClient, log))
+      .via(createDestination(eventing.out, commandHandler))
+      .dropWhile(_ => true) // TODO decide what to do with these messages, we want to keep on forever if possible
```

Comment:
- Trying to reconcile this with the idea expressed earlier (during a recent `CloudState Contributors’ Call Meeting`) regarding the following two options: (1) Produce, without _consuming_, or (2) Produce without _side-effects_. But I totally agree that `compaction`-related concerns are relatively minor at this time 👍

Reply:
@akramtexas No, that part of things is the generated IDs produced by the publishing to Pubsub (or any other messages which may be produced as the result of publication.